source/slang/slang-lower-to-ir.cpp

```diff
@@ -1047,6 +1047,11 @@ static void addLinkageDecoration(
         builder->addPublicDecoration(inst);
         builder->addKeepAliveDecoration(inst);
     }
+    if (decl->findModifier<HLSLExportModifier>())
+    {
+        builder->addHLSLExportDecoration(inst);
+        builder->addKeepAliveDecoration(inst);
+    }
```

Comment:
It isn't specific to this PR, but I definitely get worried seeing the sheer number of cases here that are adding decorations that appear to be related to linkage, but that aren't in any way tied to our current `[import(...)]` and `[export(...)]` decorations.

The `PublicModifier`/`PublicDecoration` stuff is just plain wrong, and we will need to get rid of them sooner or later in order to be able to make the `public` keyword in the language actually do anything useful.